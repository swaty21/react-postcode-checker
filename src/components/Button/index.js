import React from 'react';

const Button = ({ id, onClick, name, type = 'button' }) => {
  return (
    <button data-testid={id} onClick={onClick} name={name} type={type}>
      {name}{' '}
    </button>
  );
};

export default Button;
