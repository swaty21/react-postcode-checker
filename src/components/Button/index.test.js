/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from './index';

const props = {
  id: 'button',
  onClick: jest.fn(),
  name: 'Primary button',
  type: 'button'
};

describe('Button', () => {
  it('shoulld render Button Component', () => {
    const { container } = render(<Button {...props} />);
    expect(container).toMatchSnapshot();
  });
  it('should click on Button', () => {
    const { getByTestId } = render(<Button {...props} />);
    const btnElement = getByTestId('button');
    fireEvent.click(btnElement);
    expect(props.onClick).toHaveBeenCalled();
  });
});
