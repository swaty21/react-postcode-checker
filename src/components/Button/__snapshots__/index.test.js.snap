// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button shoulld render Button Component 1`] = `
<div>
  <button
    data-testid="button"
    name="Primary button"
    type="button"
  >
    Primary button
     
  </button>
</div>
`;
